src/controllers/taskController.js

```diff
@@ -712,7 +712,7 @@ const taskController = function (Task) {
   };
 
   const updateTask = async (req, res) => {
-    if (!(await hasPermission(req.body.requestor, 'updateTask'))) {
+    if (!(await hasPermission(req.body.requestor, 'updateTask') || await hasPermission(req.body.requestor, 'deleteDashboardTask'))) {
```

Comment:
DeleteDashboardTask should only allow the requestor to delete tasks of users on their team, not for anyone. 
Also, this looks like the wrong function, `deleteDashboardTask` should affect `deleteTask()`, not `updateTask()`.